disnake/threads.py

```diff
@@ -959,3 +1010,254 @@ def _from_data(self, data: ThreadMemberPayload):
+
+        emoji_name, emoji_id = PartialEmoji._to_name_id(emoji)
+        self._emoji_id: Optional[int] = emoji_id
+        self._emoji_name: Optional[str] = emoji_name
+
+    def __str__(self) -> str:
+        return self.name
+
+    def __eq__(self, other):
+        return (
+            isinstance(other, PartialThreadTag)
+            and self.name == other.name
+            and self.moderated == other.moderated
+            and self._emoji_id == other._emoji_id
+            and self._emoji_name == other._emoji_name
+        )
+
+    def to_dict(self) -> PartialThreadTagPayload:
+        return {
+            "name": self.name,
+            "emoji_id": self._emoji_id,
+            "emoji_name": self._emoji_name,
+            "moderated": self.moderated,
+        }
+
+
+# TODO: just `Tag` instead of `ThreadTag`?
+#       or perhaps `ForumTag` to match apidocs name?
+
+# TODO: don't inherit from `Hashable` and inherit `__eq__` from `PartialThreadTag` instead?
+#       would make things like comparing `available_tags` in auditlogs easier
+class ThreadTag(Hashable, PartialThreadTag):
+    """
+    Represents a tag for threads in forum channels.
+
+    .. container:: operations
+
+        .. describe:: x == y
+
+            Checks if two tags are equal.
+
+        .. describe:: x != y
+
+            Checks if two tags are not equal.
+
+        .. describe:: hash(x)
+
+            Returns the tag's hash.
+
+        .. describe:: str(x)
+
+            Returns the tag's name.
+
+    .. versionadded:: 2.6
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The tag's ID.
+    name: :class:`str`
+        The tag's name.
+    moderated: :class:`bool`
+        Whether only moderators can apply this tag to threads.
+    """
+
+    __slots__ = ("id", "_channel", "_state")
+
+    def __init__(
+        self,
+        *,
+        data: ThreadTagPayload,
+        # Object is used by audit logs
+        channel: Union[ForumChannel, Object],
+        state: ConnectionState,
+    ):
+        # emoji is set below
+        super().__init__(data["name"], emoji=None, moderated=data["moderated"])
+
+        self._channel: Union[ForumChannel, Object] = channel
+        self._state: ConnectionState = state
+
+        self.id: int = int(data["id"])
+        # emoji_id may be `0`, use `None` instead
+        self._emoji_id: Optional[int] = _get_as_snowflake(data, "emoji_id") or None
+        self._emoji_name: Optional[str] = data.get("emoji_name")
+
+    def __repr__(self) -> str:
+        return (
+            f"<ThreadTag id={self.id!r} name={self.name!r}"
+            f" moderated={self.moderated!r} emoji={self.emoji!r}>"
+        )
+
+    @property
+    def emoji(self) -> Optional[Union[Emoji, PartialEmoji]]:
+        """Optional[Union[:class:`Emoji`, :class:`PartialEmoji`]]: The emoji associated with this tag, if any."""
+        return PartialEmoji._from_name_id(self._emoji_name, self._emoji_id, state=self._state)
```

Comment:
wouldn't this just always return a PartialEmoji?

Reply:
`PartialEmoji._from_name_id` returns `Emoji | PartialEmoji | None`, I agree that the naming is a bit confusing. Didn't really know where else to put it, other than `disnake.utils` I suppose?

Reply:
Renamed slightly in https://github.com/DisnakeDev/disnake/pull/726/commits/d0bfc417947ef310a68518b8ab9b17b655982b14, moving them to utils would be somewhat annoying thanks to circular imports.